agent/kafka_handler.go

```diff
@@ -17,6 +26,91 @@ type kafkaHandler struct {
 	ctx   kafkaserver2.ConnectionContext
 }
 
+func (k *kafkaHandler) HandleCreateTopicsRequest(hdr *kafkaprotocol.RequestHeader, req *kafkaprotocol.CreateTopicsRequest, completionFunc func(resp *kafkaprotocol.CreateTopicsResponse) error) error {
+	resp := &kafkaprotocol.CreateTopicsResponse{
+		ThrottleTimeMs: 0,
+		Topics:         make([]kafkaprotocol.CreateTopicsResponseCreatableTopicResult, len(req.Topics)),
+	}
+
+	for tidx, topic := range req.Topics {
+		retentionTime := defaultRetentionTime
+		respConfigs := make([]kafkaprotocol.CreateTopicsResponseCreatableTopicConfigs, len(topic.Configs))
+
+		// Parse custom retention time from topic configs if provided
+		for cidx, config := range topic.Configs {
+			if *config.Name == "retention.ms" {
+				retentionMs, err := strconv.Atoi(*config.Value)
+				if err == nil {
+					retentionTime = time.Duration(retentionMs) * time.Millisecond
+				}
+			}
+
+			respConfigs[cidx] = kafkaprotocol.CreateTopicsResponseCreatableTopicConfigs{
+				Name:  config.Name,
+				Value: config.Value,
+			}
+		}
+
+		topicInfo := topicmeta.TopicInfo{
+			Name:           *topic.Name,
+			PartitionCount: int(topic.NumPartitions),
+			RetentionTime:  retentionTime,
+		}
+		err := k.agent.controller.TopicMetaManager.CreateTopic(topicInfo)
+		var errMsg string
+		if err != nil {
+			errMsg = err.Error()
+		}
+
+		info, ok := k.agent.controller.TopicMetaManager.TopicInfosByName[*topic.Name]
+		if !ok {
+			errMsg = fmt.Sprintf("topic %s not created successfully", *topic.Name)
+		}
+
+		resp.Topics[tidx] = kafkaprotocol.CreateTopicsResponseCreatableTopicResult{
+			Name:              topic.Name,
+			TopicId:           uint32ToBytes(uint32(info.ID)),
```

Comment:
Kafka topicID is not related to this ID. Leave it empty - it's not used in the version of the API we are using here

Reply:
makes sense. The response of version 5 doesn't contain topic Id.